osu.Desktop/DiscordRichPresence.cs

```diff
@@ -66,7 +71,7 @@ private void load(IAPIProvider provider, OsuConfigManager config)
                 activity.BindTo(u.NewValue.Activity);
             }, true);
 
-            ruleset.BindValueChanged(_ => updateStatus());
+            ruleset.BindValueChanged(_ => requestUserRulesetsRankings());
```

Comment:
Not sure this should be firing requests every time the ruleset changes.

Reply:
Is there another way we could get an updated version of the rankings, or a better timing to do it ?

Reply:
Oh ok, so it would be fine to only make that request part of the "initial" connection request ?

Reply:
I think so but it may be best to wait for a second opinion here.

Reply:
Also i'm not sure where to find that first connection request, is there a specific place/event that could be reach by DiscordRichPresence to know once the client has logged in ?

Reply:
You'll probably want to fire the request [on change to `LocalUser`](https://github.com/ppy/osu/blob/80fa90f65d149d806a05a2e616f047616591ae04/osu.Desktop/DiscordRichPresence.cs#L65-L72). Which, by the way, is more correct anyway, because with the way you have things the rank display could break if you logged out and then in to a different account for whatever reason.